test/extended/builds/build_no_outputname.go

```diff
@@ -20,25 +20,35 @@ var _ = g.Describe("builds: no name specified for the output image", func() {
 
 	g.Describe("building from templates", func() {
 		oc.SetOutputDir(exutil.TestContext.OutputDir)
-
 		g.It(fmt.Sprintf("should create an image from %q docker template without an output image reference defined", dockerImageFixture), func() {
 			err := oc.Run("create").Args("-f", dockerImageFixture).Execute()
 			o.Expect(err).NotTo(o.HaveOccurred())
 
-			g.By("expecting build to pass without an output image reference specified")
-			out, err := oc.Run("start-build").Args("test-docker", "--follow", "--wait").Output()
+			out, err := retryBuildOnFailure("test-docker", oc)
 			o.Expect(err).NotTo(o.HaveOccurred())
+			g.By("expecting build to pass without an output image reference specified")
 			o.Expect(out).Should(o.ContainSubstring(`Build does not have an Output defined, no output image was pushed to a registry.`))
 		})
 
 		g.It(fmt.Sprintf("should create an image from %q S2i template without an output image reference defined", s2iImageFixture), func() {
+			g.By("expecting build to pass without an output image reference specified")
 			err := oc.Run("create").Args("-f", s2iImageFixture).Execute()
 			o.Expect(err).NotTo(o.HaveOccurred())
 
-			g.By("expecting build to pass without an output image reference specified")
-			out, err := oc.Run("start-build").Args("test-sti", "--follow", "--wait").Output()
+			out, err := retryBuildOnFailure("test-sti", oc)
 			o.Expect(err).NotTo(o.HaveOccurred())
 			o.Expect(out).Should(o.ContainSubstring(`Build does not have an Output defined, no output image was pushed to a registry.`))
 		})
 	})
 })
+
+func retryBuildOnFailure(buildName string, oc *exutil.CLI) (out string, err error) {
+	for i := 0; i <= 3; i++ {
```

Comment:
maybe add some sleep here (or use the k8s Poll() func)... I don't think retrying this fast will make things better.